medusa/process_tv.py

```diff
@@ -16,670 +15,596 @@
+
+    @property
+    def output(self):
+        return '\n'.join(self._output)
+
+    def _log(self, message, level=logger.INFO):
+        logger.log(message, level)
+        self._output.append(message)
+
+    def process(self, nzb_name=None, force=False, is_priority=None, delete_on=False, failed=False,
+                proc_type='auto', ignore_subs=False):
+        """
+        Scan through the files in the root directory and process whatever media files are found.
+
+        :param nzb_name: The NZB name which resulted in this folder being downloaded
+        :param force: True to postprocess already postprocessed files
+        :param is_priority: Boolean for whether or not is a priority download
+        :param delete_on: Boolean for whether or not it should delete files
+        :param failed: Boolean for whether or not the download failed
+        :param proc_type: Type of postprocessing auto or manual
+        :param ignore_subs: True to ignore setting 'postpone if no subs'
+        """
+        if not self.directory:
+            return self.output
+
+        if app.POSTPONE_IF_NO_SUBS:
+            self._log("Feature 'postpone post-processing if no subtitle available' is enabled.")
+
+        for path in self.paths:
+
+            if not self.should_process(path, nzb_name, failed):
+                continue
 
-        try:
-            logger.log(u"Deleting folder (if it's empty): %s" % folder)
-            os.rmdir(folder)
-        except (OSError, IOError) as e:
-            logger.log(u"Warning: unable to delete folder: %s: %s" % (folder, ex(e)), logger.WARNING)
-            return False
-    else:
-        try:
-            logger.log(u"Deleting folder: " + folder)
-            shutil.rmtree(folder)
-        except (OSError, IOError) as e:
-            logger.log(u"Warning: unable to delete folder: %s: %s" % (folder, ex(e)), logger.WARNING)
-            return False
+            self.result = True
 
-    return True
+            for dir_path, filelist in self._get_files(path):
 
+                sync_files = (filename
+                              for filename in filelist
+                              if is_sync_file(filename))
+                # Don't process files if they are still being synced
+                postpone = app.POSTPONE_IF_SYNC_FILES and any(sync_files)
 
-def delete_files(processPath, notwantedFiles, result, force=False):
-    """
-    Remove files from filesystem.
+                if not postpone:
 
-    :param processPath: path to process
-    :param notwantedFiles: files we do not want
-    :param result: Processor results
-    :param force: Boolean, force deletion, defaults to false
-    """
-    if not result.result and force:
-        result.output += logHelper(u"Forcing deletion of files, even though last result was not successful", logger.DEBUG)
-    elif not result.result:
-        return
+                    self._log('\nProcessing folder: {0}'.format(dir_path), logger.DEBUG)
```

Comment:
@medariox I have ideas to clean up the manual PP logging (i had tested some of them locally but didnt want to PR until you were done with your refactors)